ironfish/src/primitives/target.ts

```diff
@@ -165,9 +160,21 @@ export class Target {
       targetDifficulty = previousBlockDifficulty - BigInt(max_change) * BigInt(multiplier)
     }
 
-    const difficulty = BigIntUtils.max(targetDifficulty, Target.minDifficulty())
+    return BigIntUtils.max(targetDifficulty, Target.minDifficulty())
+  }
 
-    return difficulty
+  /**
+   * Returns the bucket values based difference time between the current and previous block, target block mining time and free time range
+   * @param diffInSeconds different in seconds between blocks
+   * @returns the bucket number
+   */
+  static findBucket(diffInSeconds: number): number {
+    const halfBucket = Math.floor(FREEZE_TIME_IN_SECONDS / 2)
+    return (
+      Math.floor(
+        (diffInSeconds - TARGET_BLOCK_TIME_IN_SECONDS + halfBucket) / FREEZE_TIME_IN_SECONDS,
```

Comment:
`FREEZE_TIME_IN_SECONDS` and `free time range` are confusing. we're not really freezing anything. maybe something like `ALLOWED_SLIPPAGE_FROM_TARGET_TIME` ?